docs/docs/spec.md

```diff
@@ -122,6 +122,14 @@ A nullness operator is one of four values:
 -   `UNSPECIFIED`
 -   `MINUS_NULL`
 
+TODO: This is the first appearance of "operator" and there is no formal definition
```

Comment:
I've never been sure what the best way to order the sections of the spec is.... I want to introduce "nullness operators" early so that I can define "augmented types" in terms of them. But then until I define that latter term, the former definition is just kind of hanging there.

As a formal definition, I want to claim something similar: The "formal definition" of a nullness operator is either:
- that it's one of the four operators listed here, or...
- that it's defined by its effect on substitution and subtyping rules, as discussed later.

I'm not sure that a formal definition could add anything beyond that, though maybe we could foreshadow that those rules are coming?

Basically, I think that it would be valuable to give some _non-normative, informal_ definition, since we currently say _nothing_ about this fundamental concept except for discussing the four operators individually later on.

Or maybe we can get away with a formal definition along the lines of the following...?

"A nullness operator is the main way that JSpecify represents information about the nullness of a given type. There are four nullness operators:"